crates/tracing-instrumentation/src/prometheus_metrics.rs

```diff
@@ -8,20 +8,25 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use metrics_exporter_prometheus::{PrometheusBuilder, PrometheusHandle};
-use metrics_util::MetricKindMask;
+use std::sync::OnceLock;
 
 use metrics_exporter_prometheus::formatting;
-use restate_types::config::CommonOptions;
+use metrics_exporter_prometheus::{PrometheusBuilder, PrometheusHandle};
+use metrics_util::MetricKindMask;
 use tokio::task::AbortHandle;
 use tokio::time::MissedTickBehavior;
 use tracing::{debug, trace};
 
+use restate_types::config::CommonOptions;
+
+use crate::GLOBAL_NODE_ID;
+
 #[derive(Default)]
 pub struct Prometheus {
     handle: Option<PrometheusHandle>,
     upkeep_task: Option<AbortHandle>,
-    global_labels: Vec<String>,
+    static_labels: Vec<String>,
+    global_labels: OnceLock<Vec<String>>,
```

Comment:
Is there a better way to model the initialization and avoid having to pass mutable references? We could also keep a single copy behind a Mutex.